pt_miniscreen/hotspot_manager.py

```diff
@@ -118,19 +121,22 @@ def register(self, hotspot_instance, collection_id=None):
 
     def _register_thread(self, hotspot_instance):
         def run():
-            while self.update_cached_images:
-                if self.is_hotspot_overlapping(hotspot_instance):
-                    self.cached_images[
-                        hotspot_instance.hotspot
-                    ] = hotspot_instance.hotspot.image
+            def cache_new_image():
+                self.cached_images[hotspot] = hotspot.image
 
-                sleep(hotspot_instance.hotspot.interval)
+            cache_new_image()
+            while self.update_cached_images:
+                if self.active and self.is_hotspot_overlapping(hotspot_instance):
```

Comment:
This is a "busy wait" if not active - perhaps set an event on active state change?